department-of-veterans-affairs/abd-vro
# Feature - CC Endpoint for dropdown contentions

## What was the problem?

- Currently, there exists a `/contention-classification/classifier` endpoint used by VA.gov for classifying Claims For Increase. This does not, however support "NEW" type claims.

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/225

## How does this fix it?[^1]

- This PR adds a new endpoint that accepts both new and claims for increase. Previous endpoint is marked as deprecated.
- The new `/contention-classification/v2/classifier` endpoint preserves the existing lookup table for claims for increase.
- For new contentions and claims for increase that are not covered in our LUT, we use a set of mappings defined by the team [HERE](https://docs.google.com/spreadsheets/d/1A5JuYwn39mHE5Mk1HazN-mxCL2TENPeyUPHHhH10g_I/edit#gid=819850041).


## How to test this PR
- navigate to `http://localhost:8120/docs` observer `/classifier` is deprecated and greyed out
- run `pytest` from `domain-cc/cc-app`
- make POST request to `http://localhost:8120/v2/classifier`
```
{"diagnostic_code": 9999999, "claim_id": 100, "form526_submission_id": 500, "contention_text": "tuberculosis", "claim_type": "claim_for_increase"} 
```


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
